lib/src/inline_syntaxes/footnote_ref_syntax.dart

```diff
@@ -0,0 +1,32 @@
+import '../ast.dart';
+import '../charcode.dart';
+import '../inline_parser.dart';
+import 'inline_syntax.dart';
+
+class FootnoteRefSyntax extends InlineSyntax {
+  FootnoteRefSyntax() : super(r'!?\[\^(\S+?)\]');
```

Comment:
Hmm, what is the leading `!?` for?

Reply:
refine the code and not extends but directly change within `LinkSyntax`